### PR TITLE
Add option to use zlib-ng as zlib library

### PR DESCRIPTION
**Pull request by byrnHDF**
Rework compression libs CMake handling
Update to latest libaec version

### CODE REVIEW

File: CMakeFilters.cmake [hunk 1]
```diff
@@ -9,14 +9,23 @@
 # If you do not have access to either file, you may request a copy from
 # help@hdfgroup.org.
 #
-option (USE_LIBAEC_STATIC "Use static AEC library " OFF)
+option (USE_ZLIB_NG "Use zlib-ng library as zlib library" OFF)
```
Comment (jhendersonHDF) [inline CMakeFilters.cmake]: Seems like we should probably be prefixing these options as in `HDF5_USE_ZLIB_NG`

Comment (byrnHDF) [reply CMakeFilters.cmake]: Yes - good point

**Review by lrknox — CHANGES_REQUESTED** [review]
Libaec 1.0.6 should still work?

**Comment by byrnHDF** [comment]
> Libaec 1.0.6 should still work?

Need to test but yes it should

**Review by lrknox — APPROVED** [review]

File: release_docs/INSTALL_CMake.txt [hunk 2]
```diff
@@ -531,12 +531,18 @@ These five steps are described in detail below.
             set (ZLIB_USE_LOCALCONTENT ON CACHE BOOL "Use local file for ZLIB FetchContent" FORCE)
```
Comment (tbeu) [inline release_docs/INSTALL_CMake.txt:531]: Why is zlib not updated to v1.3.1 on the fly?

Comment (byrnHDF) [reply release_docs/INSTALL_CMake.txt:531]: We want to use the latest release after testing.

Comment (tbeu) [reply release_docs/INSTALL_CMake.txt:531]: OK. Why was libaec updated in this PR? Seems unrelated to zlib-ng.

Comment (byrnHDF) [reply release_docs/INSTALL_CMake.txt:531]: Because I have a bad habit of adding something I think is related.

**Review by derobins — APPROVED** [review]

**Pull request merged**